nerfstudio/field_components/encodings.py

```diff
@@ -226,11 +236,16 @@ def forward(
         """
         scaled_in_tensor = 2 * torch.pi * in_tensor  # scale to [0, 2pi]
         scaled_inputs = scaled_in_tensor @ self.b_matrix  # [..., "num_frequencies"]
+        freqs = 2 ** torch.linspace(self.min_freq, self.max_freq, self.num_frequencies).to(in_tensor.device)
```

Comment:
Please do not allocate on cpu and then move to GPU. Allocate directly on target device with device=... kwargs.

Also, do you think it is better to have freqs here, or would it be better to move them to b_matrix?

Reply:
Alright, I had just copied the line from `NeRFEncoding` https://github.com/nerfstudio-project/nerfstudio/blob/e0f0cd05aa06bdd148a20ed62271d49ff0a6ac9d/nerfstudio/field_components/encodings.py#L156.
I believe it is better to have them here as otherwise the diagonal matrix calculation unnecessary happens on a larger b_matrix but I'm not completely sure.

Reply:
Sure, I agree